src/core/execute.c

```diff
@@ -411,7 +411,8 @@ static int fixup_output(ExecOutput std_output, int socket_fd) {
 static int setup_input(
                 const ExecContext *context,
                 const ExecParameters *params,
-                int socket_fd) {
+                int socket_fd,
+                int *named_iofds) {
```

Comment:
I think it would be clearer to use `int named_iofds[3]` here. It helps the reader see that this is an array, not a pointer to a fd.